src/app/qgisapp.cpp

```diff
@@ -16383,6 +16383,11 @@ void QgisApp::readProject( const QDomDocument &doc )
       mapCanvas->readProject( doc );
       views << mapCanvas;
     }
+
+    if ( QgsProject::instance()->viewSettings()->canvasUseProjectExtent() )
+    {
+      zoomFull();
```

Comment:
I would not add this here, and leave the handling to be done in QgsMapCanvas (see my other comment)